agent/stats/engine_test.go

```diff
@@ -54,6 +54,10 @@ func (engine *MockTaskEngine) ListTasks() ([]*api.Task, error) {
 	return nil, nil
 }
 
+func (engine *MockTaskEngine) GetTaskByArn(arn string) (*api.Task, bool) {
```

Comment:
Nit: is there any particular reason why you're returning a `bool` type here instead of just using `error`?